From 3d16b95acde1450d4e6a403225b79e4ead1dbf81 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:03:50 -0400
Subject: [usb] bcma: replace dma_set_mask()+dma_set_coherent_mask() with new helper

Message-id: <1410887463-20674-159-git-send-email-dzickus@redhat.com>
Patchwork-id: 95128
O-Subject: [RHEL7 PATCH 158/591] DMA-API: usb: bcma: replace dma_set_mask()+dma_set_coherent_mask() with new helper
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit d288059e0aedcee0d55074b81ef286c7f255e146
Author: Russell King <rmk+kernel@arm.linux.org.uk>
Date:   Wed Jun 26 23:49:11 2013 +0100

    DMA-API: usb: bcma: replace dma_set_mask()+dma_set_coherent_mask() with new helper

    Replace the following sequence:

     dma_set_mask(dev, mask);
     dma_set_coherent_mask(dev, mask);

    with a call to the new helper dma_set_mask_and_coherent().

    Signed-off-by: Russell King <rmk+kernel@arm.linux.org.uk>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/host/bcma-hcd.c b/drivers/usb/host/bcma-hcd.c
index df13d42..205f4a3 100644
--- a/drivers/usb/host/bcma-hcd.c
+++ b/drivers/usb/host/bcma-hcd.c
@@ -227,8 +227,7 @@ static int bcma_hcd_probe(struct bcma_device *dev)
 
  /* TODO: Probably need checks here; is the core connected? */
 
- if (dma_set_mask(dev->dma_dev, DMA_BIT_MASK(32)) ||
-     dma_set_coherent_mask(dev->dma_dev, DMA_BIT_MASK(32)))
+ if (dma_set_mask_and_coherent(dev->dma_dev, DMA_BIT_MASK(32)))
   return -EOPNOTSUPP;
 
  usb_dev = kzalloc(sizeof(struct bcma_hcd_device), GFP_KERNEL);
-- 
1.7.1